public/actions.goal.test.js

```diff
@@ -30,6 +30,61 @@ test('can add goal', (t) => {
   }));
 });
 
+test('can add multiple goals at once', (t) => {
+  const goalTextToAdd = 'foo;bar;wow';
+  const expectedGoals = ['foo', 'bar', 'wow'];
+  const initialState = {
+    goals: [],
+  };
+
+  const [state, effect] = actions.AddMultipleGoals(initialState, goalTextToAdd);
+
+  t.like(state.goals[0], {
+    text: expectedGoals[0], completed: false,
+  });
+  t.is(state.goal, '');
+
+  const stateGoalsTextArray = state.goals.map(goal => goal.text);
+  t.deepEqual(stateGoalsTextArray, expectedGoals);
```

Comment:
This could probably be condensed to:

```js
t.is(state.goals.length, expectedGoals.length);
expectedGoals.forEach((text, index) => {
  t.like(state.goals[index], { text, completed: false });
});
```

Reply:
Looks like a similar pattern could work on the rest of the tests, too

Reply:
We should also be able to do an effect assertion, which might look something like:

```js
t.deepEqual(effect, effects.UpdateGoals({
  websocket,
  goals: state.goals,
});
```

Which means you'll have to define a websocket/put it in your initial state, but for the sake of the test, it just needs to be an empty `{}` object.

Reply:
Added.  Hm, that's a really nice benefit to the design.  Very testable code.